### PR TITLE
Swift 3.0 Migration

### PR DESCRIPTION
**Pull request by sleroux**
WIP for Swift 3.0 migration.

Current strategy is to go piece by piece through each framework/target and use the migrator which so far has been pretty solid.

### CODE REVIEW

**Comment by varkor** [comment]
All of the internal code has no been migrated to Swift 3.0. Now, we just need to update our dependencies to Swift 3.0 versions.

**Comment by farhanpatel** [comment]
woo! Thats an impressive number of changes. I'm curious, was the migrator tool helpful? Or did you have to manually convert things?

**Comment by varkor** [comment]
The migration assistant does lots of the heavy lifting, but there are a couple of areas it stumbles with (some name collisions, certain places optionals aren't being handled in the best way, some random issues it didn't seem to pick up on). Also, the naming convention for methods has changed with Swift to favour explicit named first parameters over overly-descriptive method names, so there was quite a lot of refactoring too. There are a lot of methods, though, so there are probably a few that slipped through the net.

**Comment by sleroux** [comment]
Hey @varkor thanks a ton for picking this up! There were a ton of changes needed to be made in the Client/Sync targets and that was no small task. Really happy to have these changes made so we can just wait for the dependencies to update instead of the other way around!

**Comment by farhanpatel** [comment]
Dependencies that still need to migrate to 3.0
- [ ]  [Sqlite.swift](https://github.com/stephencelis/SQLite.swift/issues/455)

**Comment by varkor** [comment]
I've taken a look through the first few commits, and made some changes to hopefully make the method names more Swift 3.0-y. Could do with someone looking through at some change to make sure they all seem sensible, though.

**Comment by bytegh** [comment]
Hey, guys:

so I we start to use this PR in my project(migrated to swift3.0 already). Or I still need to wait? 

thanks! this is a amazing library :)

**Comment by skreutzberger** [comment]
Why don’t you guys switch to [SwiftyBeaver](https://github.com/SwiftyBeaver/SwiftyBeaver) for logging on debug and release? 

Besides being more actively developed with a larger, more active userbase, it is also supporting the latest Swift 3 GM. This week the swift3 branch will be even merged into master and will become the new standard.

Disclaimer:  I am the author 😬

**Comment by rajmohangoaugment** [comment]
is the swift 3.0 is in progress or not happening anytime soon ? I would prefer to wait if its happening since i may avoid duplication of conversion work.

File: Client/Frontend/Browser/FindInPageBar.swift [hunk 1]
```diff
@@ -56,43 +56,43 @@ class FindInPageBar: UIView {
...
 
         matchCountView.textColor = FindInPageUX.MatchCountColor
         matchCountView.font = FindInPageUX.MatchCountFont
-        matchCountView.hidden = true
+        matchCountView.isHidden = true
         addSubview(matchCountView)
 
-        previousButton.setImage(UIImage(named: "find_previous"), forState: UIControlState.Normal)
-        previousButton.setTitleColor(FindInPageUX.ButtonColor, forState: UIControlState.Normal)
+        previousButton.setImage(UIImage(named: "find_previous"), for: UIControlState())
```
Comment (fluffyemily) [inline Client/Frontend/Browser/FindInPageBar.swift]: This should be `UIControlState.normal` not UIControlState() as should the other below

**Pull request closed without merging**